wsdl/types.go

```diff
@@ -236,10 +236,16 @@ type IO struct {
 
 // Binding describes SOAP to WSDL binding.
 type Binding struct {
-	XMLName    xml.Name            `xml:"binding"`
-	Name       string              `xml:"name,attr"`
-	Type       string              `xml:"type,attr"`
-	Operations []*BindingOperation `xml:"operation"`
+	XMLName     xml.Name            `xml:"binding"`
+	Name        string              `xml:"name,attr"`
+	Type        string              `xml:"type,attr"`
+	SOAPBinding *SOAPBinding        `xml:"binding"`
+	Operations  []*BindingOperation `xml:"operation"`
+}
+
+type SOAPBinding struct {
```

Comment:
This is a such a small diff and yet it cannot figure out the difference between Binding and SOAPBinding by just looking at it. Also this will fail golint due to exported type with no documentation.